app/src/main/java/com/example/androidtrainingproject/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,167 @@
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_filter_list),
+                    contentDescription = null,
+                    modifier = Modifier.height(dimensionResource(id = R.dimen.icon_large))
+                )
+            }
+
+            if (homeViewModel.isLoading) {
+                Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
+                    CircularProgressIndicator()
+                }
+            }
+
+            homeViewModel.filteredProducts.let { products ->
+                LazyColumn(modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f)) {
+                    items(products) { product ->
+                        Row(
+                            modifier = Modifier
+                                .padding(vertical = dimensionResource(id = R.dimen.padding_sm_plus)),
+
+                        ) {
+                            AsyncImage(
+                                model = product.image,
+                                modifier = Modifier
+                                    .width(dimensionResource(id = R.dimen.home_image_size))
+                                    .clip(RoundedCornerShape(dimensionResource(id = R.dimen.padding_sm_plus)))
+                                    .clickable(onClick = {
+                                        navController.navigate("productDetails/${product.id}")
+                                    }),
+                                contentScale = ContentScale.Fit,
+                                alignment = Alignment.Center,
+                                contentDescription = stringResource(id = R.string.product_image_description)
+                            )
+                            Column(
+                                modifier = Modifier.padding(start = dimensionResource(id = R.dimen.padding_m))
+                            ) {
+                                Text(
+                                    text = stringResource(id = R.string.product_category_label) + " " + product.category,
+                                    fontSize = dimensionResource(id = R.dimen.font_size_sm).value.sp,
+                                    color = LabelText
+                                )
+                                Text(
+                                    text = product.title,
+                                    fontWeight = FontWeight.Bold
+                                )
+                                Text(
+                                    text = product.short_description,
+                                    fontSize = dimensionResource(id = R.dimen.font_size_sm).value.sp,
+                                    lineHeight = dimensionResource(id = R.dimen.home_custom_line_height).value.sp,
+                                    color = LabelText
+                                )
+                                Row(
```

Comment:
Better extract to a separate composable. Might be in the same file for such a small layout but might as well be in a separate file in an adjacent `layouts` folder.